perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/VosManagerBlImpl.java

```diff
@@ -902,14 +928,54 @@ public boolean usesEmbeddedGroupRegistrations(PerunSession sess, Vo vo) {
 
 	@Override
 	public void addMemberVo(PerunSession sess, Vo vo, Vo memberVo) throws RelationExistsException {
-		// todo - add necessary logic for adding new member vo
 		vosManagerImpl.addMemberVo(sess, vo, memberVo);
+
+		List<Member> memberVoMembers = perunBl.getMembersManagerBl().getMembers(sess, memberVo);
+
+		for (Member member : memberVoMembers) {
+			try {
+
+				try {
+					// if user is member in both vos, update memberOrganizations list attribute
+					Member existingMember = perunBl.getMembersManagerBl().getMemberByUserId(sess, vo, member.getUserId());
```

Comment:
Maybe we could check his current status and try to validate him if he is not valid yet. And perhaps also remove his expiration time if it is set for him because the member should follow the parent VOs lifecycle only if he is not valid in any member organization.